overlord/snapstate/handlers_link_test.go

```diff
@@ -722,6 +760,64 @@ func (s *linkSnapSuite) TestDoUndoUnlinkCurrentSnapCore(c *C) {
 	c.Check(s.stateBackend.restartRequested, DeepEquals, []state.RestartType{state.RestartDaemon})
 }
 
+func (s *linkSnapSuite) TestDoUndoUnlinkCurrentSnapCoreBase(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+
+	r := snapstatetest.MockDeviceModel(ModelWithBase("core18"))
+	defer r()
+
+	s.fakeBackend.linkSnapMaybeReboot = true
+
+	s.state.Lock()
+	defer s.state.Unlock()
+	// we need to init the boot-id
+	err := s.state.VerifyReboot("some-boot-id")
+	c.Assert(err, IsNil)
+
+	si1 := &snap.SideInfo{
+		RealName: "core18",
+		Revision: snap.R(1),
+	}
+	si2 := &snap.SideInfo{
+		RealName: "core18",
+		Revision: snap.R(2),
+	}
+	snapstate.Set(s.state, "core18", &snapstate.SnapState{
+		Sequence: []*snap.SideInfo{si1},
+		Current:  si1.Revision,
+		Active:   true,
+		SnapType: "base",
+	})
+	t := s.state.NewTask("unlink-current-snap", "test")
+	t.Set("snap-setup", &snapstate.SnapSetup{
+		SideInfo: si2,
+	})
+	chg := s.state.NewChange("dummy", "...")
+	chg.AddTask(t)
+
+	terr := s.state.NewTask("error-trigger", "provoking total undo")
+	terr.WaitFor(t)
+	chg.AddTask(terr)
+
+	s.state.Unlock()
+	for i := 0; i < 3; i++ {
```

Comment:
I see other tests in this file follow the same pattern of ensuring and waiting 3 times, but I was just wondering is there a reason why 3? Just to be extra, extra, extra sure that stuff happens? Not a blocker for this PR obviously

Reply:
there are two tasks involved (with the erroring one) at least plus a cleanup phase I think, so 3 is not far from minimal I think